Add configurable list of classmethod / staticmethod decorators

saifelse opened the pull request:
- Add classmethod-decorator and staticmethod-decorator config parameters
- Add support for parsing CLI-style flake8-config parameters in testing
  framework.
- Remove support for multiple `codes` in a testcase, since it was never used.
- Add tests for late-decorated methods

Fixes #38

sigmavirus24 commented on pep8ext_naming.py:
```diff
@@ -98,11 +120,35 @@ def add_options(cls, parser):
                      help='List of names the pep8-naming plugin should '
                           'ignore. (Defaults to %default)')
 
+        classmethod_decorators = ','.join(_default_classmethod_decorators)
```
This should be sorted so the output is reproducible.

sigmavirus24 commented on pep8ext_naming.py:
```diff
@@ -98,11 +120,35 @@ def add_options(cls, parser):
+                     help='List of method decorators pep8-naming plugin '
+                          'should consider classmethod (Defaults to %default)')
+
+        staticmethod_decorators = ','.join(_default_staticmethod_decorators)
```
This should also be sorted.

sigmavirus24 commented on run_tests.py:
```diff
@@ -1,14 +1,21 @@
+import sys
+
+import flake8
+from flake8.options.manager import OptionManager
```
You should not be importing this. It's a private implementation detail of Flake8 that may change location.

saifelse commented:
> This is a fantastic start @saifelse

@sigmavirus24: Thanks! Note that the latest update is broken, but works after applying [a bugfix for flake8_polyfill](https://gitlab.com/pycqa/flake8-polyfill/merge_requests/4)

saifelse commented:
@sigmavirus24: just checking in to see if you've had a chance to look at this PR / [bugfix for flake8_polyfill](https://gitlab.com/pycqa/flake8-polyfill/merge_requests/4). Cheers.

saifelse commented:
@sigmavirus24 : checking in again to see if you've had a chance to look this over or the flake8_polyfill bugfix. Thanks!

sigmavirus24 commented:
Hey @saifelse I'm truly sorry it took me so long to work my way back around to this project/PR. The changes here look excellent. Can you rebase them? If not, just let me know and I'll tackle that myself.

saifelse commented:
Hey @sigmavirus24, no worries! I rebased and fixed the merge conflicts.

Note that this is dependent on flake8-polyfill being republished for which I've opened a merge request: https://gitlab.com/pycqa/flake8-polyfill/merge_requests/6

sigmavirus24 commented:
Closing and then re-opening to test against 1.0.2 of flake8-polyfill
